R/fit_best.R

```diff
@@ -100,7 +104,7 @@ fit_best.tune_results <- function(x,
     if (is.null(metric)) {
       metric <- .get_tune_metric_names(x)[1]
```

Comment:
If `eval_time = NULL`, we want to choose the first one used in the `tune_results` object and document that, rather than pass it on directly to `show_best()` because that will issue a warning. This is to handle `eval_time` analogous to how we handle `metric` here for `fit_best()`.

https://github.com/tidymodels/tune/issues/766#issuecomment-1822971743

Reply:
See changes: It will silently use the first given time for dynamic metrics and passes NULL otherwise.